pkg/sql/opt/norm/prune_cols_funcs.go

```diff
@@ -184,6 +184,16 @@ func neededMutationFetchCols(
 		}
 	}
 
+	// System columns might have been added into the needed fetch col set from the
+	// primary index. These columns are implicit and should not be part of the set
+	// of fetch columns, so we remove them here.
+	table := tabMeta.Table
+	for i := 0; i < table.ColumnCount(); i++ {
```

Comment:
Because of the API change, there isn't an easy way to iterate over only the system columns of a table. I'm not sure that is a big deal though.